src/karts/abstract_kart.cpp

```diff
@@ -79,6 +99,12 @@ void AbstractKart::reset()
 }   // reset
 
 // ----------------------------------------------------------------------------
+const AbstractCharacteristic* AbstractKart::getCharacteristic() const
+{
+    return &(*m_characteristic);
```

Comment:
Can you use http://en.cppreference.com/w/cpp/memory/unique_ptr/get please? I see the same deference in other places where you used unique_ptr.